keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -235,10 +371,26 @@ def generate(
             # backward compat.
             sampler.jit_compile = self.jit_compile
         sampler.run_eagerly = self.run_eagerly
+        initial_output, cache = None, None
+        x, _, _ = self.preprocessor(prompt)
```

Comment:
Another thing I just noticed, this is assuming that the preprocessor has a wider sequence length than max_length, I'm not sure we should make that assumption, as `preprocessor` is passed in by the user.

Reply:
I think it's okay when preprocessor has a shorter seq length. The cache's length is `max(max_length, gpt2_seq_len)`, if the preprocessor has a `seq_len < max_length < gpt2_seq_len`, then inside the `sampler` call, the prompt will be padded to `max_length`. If `seq_len > gpt2_seq_len`, the forward pass is broken anyway.